File: src/Java/greedy/45.跳跃游戏-ii.java
/*
 * @lc app=leetcode.cn id=45 lang=java
 *
 * [45] 跳跃游戏 II
 */

// @lc code=start
class Solution {
    public int jump(int[] nums) {
        int end = 0, maxPosition = 0, ans = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            maxPosition = Math.max(maxPosition, nums[i] + i);
            if (end == i) {
                ans++;
                end = maxPosition;
            }
        }
        return ans;
    }
}
// @lc code=end

File: src/Python/array/45.跳跃游戏-ii.py
#
# @lc app=leetcode.cn id=45 lang=python
#
# [45] 跳跃游戏 II
#

# @lc code=start
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        end = 0
        ans = 0
        maxPosition = 0
        for i in range(len(nums) - 1):
            maxPosition = max(nums[i] + i, maxPosition)
            if (end == i):
                ans += 1
                end = maxPosition
        return ans
# @lc code=end

